protocol/messenger_handler.go

```diff
@@ -1594,21 +1595,29 @@ func (m *Messenger) HandleCommunityRequestToJoinResponse(state *ReceivedMessageS
 		return ErrInvalidCommunityID
 	}
 
-	updatedRequest, err := m.communitiesManager.HandleCommunityRequestToJoinResponse(signer, requestToJoinResponseProto)
+	myCancelledRequestToJoin, err := m.MyCanceledRequestToJoinForCommunityID(requestToJoinResponseProto.CommunityId)
+
 	if err != nil {
 		return err
 	}
 
-	if updatedRequest != nil {
-		state.Response.RequestsToJoinCommunity = append(state.Response.RequestsToJoinCommunity, updatedRequest)
+	// Updating the request if request it was cancelled breaks the test
+	if myCancelledRequestToJoin == nil {
```

Comment:
Since `m.MyCanceledRequestToJoinForCommunityID` returns an error if there are no rows, the `if myCancelledRequestToJoin == nil {` will never be reached in that case, is that right?

Reply:
You were right, Fixed.